pkgs/by-name/cm/cmake/package.nix

```diff
@@ -181,6 +181,14 @@ stdenv.mkDerivation (finalAttrs: {
     sed -i 's|bin/cmake|${buildPackages.cmakeMinimal}/bin/cmake|g' Makefile
   '';
 
+  # Undo some of `fixCmakeFiles` for Darwin to make sure that checks libraries in the SDK find them
+  # (e.g., `find_library(MATH_LIBRARY m)` should find `$SDKROOT/usr/lib/libm.tbd`).
+  postFixup = lib.optionalString stdenv.hostPlatform.isDarwin ''
+    substituteInPlace "$out/share/cmake-${lib.versions.majorMinor finalAttrs.version}/Modules/Platform/Darwin.cmake" \
+       --replace-fail '/var/empty/include' '/usr/include' \
+       --replace-fail '/var/empty/lib' '/usr/lib'
+  '';
```

Comment:
It’s needed when targeting Darwin, which conceivably includes from non-Darwin to Darwin cross.

Reply:
My understanding is `targetPlatform` should never be used unless your package is a compiler. Some of the proposals to rework cross-compilation actually remove it except for compilers. I’m also inclined to stick with with `hostPlatform` because every time I try to use something else, I usually end up being wrong about it. 😅

Reply:
Yeah, fair enough.